src/lib/draw-and-pick.js

```diff
@@ -87,27 +87,53 @@ export function pickLayers(gl, {
     });
 
     // Read color in the central pixel, to be mapped with picking colors
-    const color = new Uint8Array(4);
-    gl.readPixels(deviceX, deviceY, 1, 1, GL.RGBA, GL.UNSIGNED_BYTE, color);
+    const pickedColor = new Uint8Array(4);
+    gl.readPixels(deviceX, deviceY, 1, 1, GL.RGBA, GL.UNSIGNED_BYTE, pickedColor);
 
     // restore blend mode
     setBlendMode(gl, oldBlendMode);
     // Picking process end
 
     // Process picked info start
     // Decode alpha to layer index
-    const pickedLayer = layers[color[3] - 1];
+    const pickedLayer = pickedColor[3] ? layers[pickedColor[3] - 1] : null;
+    const affectedLayers = [pickedLayer];
+
+    if (mode === 'hover') {
+      // only invoke onHover events if picked object has changed
+      const {lastPickedColor, lastPickedLayerId} = context;
+
+      const pickedLayerChanged = (pickedLayer && pickedLayer.props.id) !== lastPickedLayerId;
+      const pickedColorChanged = (lastPickedColor[0] !== pickedColor[0] ||
+        lastPickedColor[1] !== pickedColor[1] || lastPickedColor[2] !== pickedColor[2]);
+
```

Comment:
We could store a `pickedLayerIndex` instead of a `pickedLayerColor` so we didn't have to work with color components.